bigchaindb/common/transaction.py

```diff
@@ -42,6 +44,51 @@
 )
 
 
+def memoize(func):
+    cache = func.cache = {}
+
+    @functools.wraps(func)
+    def memoized_func(*args, **kwargs):
+        key = args[1]['id']
+        if key not in cache:
```

Comment:
is there a timeout on this cache?

Reply:
This is not a production ready cache. Ideally, we would have a fixed size LRU cache.

Reply:
This is a good approach to independently cache each 'key', otherwise LRU cache doesn't simply allow hashing dict/lists.